e2e/machineset_migration_test.go

```diff
@@ -213,13 +214,676 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
+					if err != nil {
+						return nil, corev1.ErrIntOverflowGenerated
+					}
+					return capiMachine, nil
+				}, capiframework.WaitLong, capiframework.RetryLong).Should(SatisfyAll(
+					HaveField("Status.V1Beta2.Conditions", ContainElement(SatisfyAll(
+						HaveField("Type", Equal(CAPIPausedCondition)),
+						HaveField("Status", Equal(metav1.ConditionTrue)),
+						HaveField("Reason", Equal("Paused")),
+					))),
+				))
+			})
+
+			It("should switch AuthoritativeAPI to ClusterAPI succeed", func() {
+				updateMachineSetAuthoritativeAPI(mapiMachineSet, mapiv1beta1.MachineAuthorityClusterAPI, mapiv1beta1.MachineAuthorityClusterAPI)
+				verifySynchronizedCondition(mapiMachineSet, mapiv1beta1.MachineAuthorityClusterAPI)
+				verifyMAPIPausedCondition(mapiMachineSet, mapiv1beta1.MachineAuthorityClusterAPI)
+				verifyCAPIPausedCondition(capiMachineSet, mapiv1beta1.MachineAuthorityClusterAPI)
+			})
+
+			It("should scale up CAPI MachineSet to 3 succeed after switching AuthoritativeAPI to ClusterAPI", func() {
+				By("Scaling up CAPI MachineSet to 3")
+				Expect(capiframework.ScaleMachineSet(mapiMSAuthMAPIName, 3, capiframework.CAPINamespace)).To(Succeed(), "should be able to scale up CAPI MachineSet")
+
+				By("Verifying a new CAPI Machine is running and Paused condition is False")
+				Eventually(func() (*capiv1beta1.Machine, error) {
+					capiMachine, err := capiframework.GetNewestMachineFromMachineSet(cl, capiMachineSet)
+					if err != nil {
+						return nil, err
+					}
+					return capiMachine, nil
+				}, capiframework.WaitLong, capiframework.RetryLong).Should(SatisfyAll(
+					HaveField("Status.Phase", HaveValue(Equal(string(capiv1beta1.MachinePhaseRunning)))),
+					HaveField("Status.V1Beta2.Conditions", ContainElement(SatisfyAll(
+						HaveField("Type", Equal(CAPIPausedCondition)),
+						HaveField("Status", Equal(metav1.ConditionFalse)),
+						HaveField("Reason", Equal("NotPaused")),
+					))),
+				))
+
+				By("Verifying there is a non-authoritative, paused MAPI Machine mirror for the new CAPI Machine")
+				Eventually(func() (*mapiv1beta1.Machine, error) {
+					mapiMachine, err := mapiframework.GetLatestMachineFromMachineSet(ctx, cl, mapiMachineSet)
+					if err != nil {
+						return nil, err
+					}
+					return mapiMachine, nil
+				}, capiframework.WaitShort, capiframework.RetryShort).Should(SatisfyAll(
+					HaveField("Status.AuthoritativeAPI", Equal(mapiv1beta1.MachineAuthorityClusterAPI)),
+					HaveField("Status.Conditions", ContainElement(SatisfyAll(
+						HaveField("Type", Equal(MAPIPausedCondition)),
+						HaveField("Status", Equal(corev1.ConditionTrue)),
+						HaveField("Reason", Equal("AuthoritativeAPINotMachineAPI")),
+					))),
+				))
+
+				By("Verifying old Machines still exist and authority on them is still MachineAPI")
+				Eventually(komega.Object(firstMAPIMachine)).Should(HaveField("Status.AuthoritativeAPI", Equal(mapiv1beta1.MachineAuthorityMachineAPI)))
```

Comment:
Should we put timeouts on these eventually calls? So we stay consistent thoughout the file?